plugins/plugins.go

```diff
@@ -63,19 +68,86 @@ func (m *Manager) Register(plugin Plugin) {
 	m.plugins = append(m.plugins, plugin)
 }
 
+func (m *Manager) GetCompiler() *ast.Compiler {
+	m.mtx.RLock()
+	defer m.mtx.RUnlock()
+	return m.Compiler
+}
+
+func (m *Manager) setCompiler(compiler *ast.Compiler) {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+	m.Compiler = compiler
+}
+
 // Start starts the manager.
 func (m *Manager) Start(ctx context.Context) error {
 	if m == nil {
 		return nil
 	}
+
+	// open transaction
+	txn, err := m.Store.NewTransaction(ctx, storage.WriteParams)
```

Comment:
Two comments here.

1. You can use the storage.Txn helper here. If the transaction fails, just return the error as it indicates OPA was started with bad policies.

2. This transaction doesn't have to do any writes.

Reply:
Done